src/app/compare/compare-tree/compare-tree.component.ts

```diff
@@ -1,209 +1,156 @@
-    if (parent && parent.value.name == path[0]) {
-      path.shift();
-      return this.checkIfSameParents(parent, parent.parentId, path);
     }
-    return false;
+    this.compareEvent.emit();
   }
 
-  createTrees(viewName: string, nodeLinkStrategy: NodeLinkStrategy, leftReport: Report, rightReport: Report) {
-    this.viewName = viewName;
-    this.nodeLinkStrategy = nodeLinkStrategy;
-    let nls = localStorage.getItem(this.viewName + '.NodeLinkStrategy');
-    if (nls) {
-      this.nodeLinkStrategy = NodeLinkStrategy[nls as keyof typeof NodeLinkStrategy];
+  selectByCheckPointNumber(tree: NgSimpleFileTree, treeItem: Report | Checkpoint): void {
+    if (ReportUtil.isReport(treeItem)) {
+      tree.selectItem(tree.getItems()[0].path);
+    } else if (ReportUtil.isCheckPoint(treeItem)) {
+      this.selectCheckpoint(tree, tree.getItems()[0], treeItem);
     }
-    const leftTree = this.helperService.convertReportToJqxTree(leftReport);
-    const rightTree = this.helperService.convertReportToJqxTree(rightReport);
-    const both = this.iterateToMakeLabelsRed(leftTree, rightTree);
-
-    this.leftTreeReference.createComponent({
-      height: '100%',
-      source: [both.left],
-    });
-    this.rightTreeReference.createComponent({
-      height: '100%',
-      source: [both.right],
-    });
-
-    this.selectFirstItem();
-  }
-
-  selectFirstItem() {
-    this.leftReport = this.leftTreeReference.getItems()[0];
-    this.rightReport = this.rightTreeReference.getItems()[0];
-    this.leftTreeReference.selectItem(this.leftReport);
-    this.rightTreeReference.selectItem(this.rightReport);
-    this.compareEvent.emit({
-      leftReport: this.leftReport,
-      rightReport: this.rightReport,
-    });
-  }
-
-  makeLabelsRed(left: any, right: any) {
-    this.redLabel(left);
-    this.redLabel(right);
-  }
-
-  redLabel(item: any) {
-    item.label = `<span style='color: red;'>${item.label}</span>`;
   }
 
-  iterateToMakeLabelsRed(leftItem: any, rightItem: any) {
-    let result = this.checkIfLabelsDifferent(leftItem, rightItem);
-    const shortestTreeLength = Math.min(leftItem.items.length, rightItem.items.length);
-    this.makeRestOfTreesRed(shortestTreeLength, rightItem.items, leftItem.items);
-
-    for (let i = 0; i < shortestTreeLength; i++) {
-      let both = this.iterateToMakeLabelsRed(leftItem.items[i], rightItem.items[i]);
-      leftItem.items[i] = both.left;
-      rightItem.items[i] = both.right;
+  selectCheckpoint(tree: NgSimpleFileTree, item: FileTreeItem, checkpointToMatch: Checkpoint): void {
+    if (item.children) {
+      for (const child of item.children) {
+        const checkpoint = child.originalValue as Checkpoint;
+        if (this.getCheckpointId(checkpoint.uid) === this.getCheckpointId(checkpointToMatch.uid)) {
+          tree.selectItem(child.path);
+          return;
+        } else {
+          this.selectCheckpoint(tree, child, checkpointToMatch);
+        }
```

Comment:
Why return on the end of the if and also have an else?

Reply:
Because it's in a for loop

Reply:
Ah my bad!